hydra-cluster/src/HydraNode.hs

```diff
@@ -281,6 +283,73 @@ withConfiguredHydraCluster tracer workDir nodeSocket firstNodeId allKeys hydraKe
         hydraScriptsTxId
         (\c -> startNodes (c : clients) rest)
 
+withOfflineHydraNode ::
+  Tracer IO HydraNodeLog ->
+  OfflineConfig ->
+  FilePath ->
+  Int ->
+  SigningKey HydraKey ->
+  (HydraClient -> IO a) ->
+  IO a
+withOfflineHydraNode tracer offlineConfig workDir hydraNodeId hydraSKey action =
+  withLogFile logFilePath $ \logFileHandle -> do
+    withOfflineHydraNode' offlineConfig workDir hydraNodeId hydraSKey (Just logFileHandle) $ do
+      \_stdoutHandle _stderrHandle processHandle -> do
+        result <-
+          race
+            (checkProcessHasNotDied ("hydra-node (" <> show hydraNodeId <> ")") processHandle)
+            (withConnectionToNode tracer hydraNodeId action)
+        case result of
+          Left e -> absurd e
+          Right a -> pure a
+ where
+  logFilePath = workDir </> "logs" </> "hydra-node-" <> show hydraNodeId <.> "log"
+
+withOfflineHydraNode' ::
```

Comment:
I don't think we need this separation as `withOfflineHydraNode'` is used in only one place. And even the other 2 functions are clunky and should be refactored.

Reply:
I'll cover this in a refactor PR.